### PR TITLE
fix: payment encoder for ERC777-stream payment network

### PR DESCRIPTION
**Pull request by bertux**
## Description of the changes
Adding support for ERC777-stream payment network in payment encoder.

### CODE REVIEW

**Comment by bertux** [comment]
Not needed until we can pay 2 invoices with the same SuperToken to the same address.
Will need to see it with @leoslr

**Comment by bertux** [comment]
waiting for review from @leoslr to see if Signer can be used instead of Provider.

**Review by KolevDarko — APPROVED** [review]

File: packages/payment-processor/src/payment/encoder-payment.ts
Comment (yomarion) [inline packages/payment-processor/src/payment/encoder-payment.ts:33]: NOK for this breaking change. I understand where it comes from but I think it's not worth it. You should instead, as a first step at least, add an equivalent of `encodeRequestPaymentWithoutSwap`, async and specific to ERC777, and handle this case separately in your back-end.

**Review by yomarion — CHANGES_REQUESTED** [review]
We cannot change one function signature just for one integration; it's too unsure and quite impactful. If we want to unify the logic, we should question the asynchronous aspect of building a transaction; in the end, is the SDK suited for what we do?

**Review by leoslr — APPROVED** [review]
Good, maybe just add comments for the superfluid functions.

File: packages/payment-processor/src/payment/encoder-payment.ts
Comment (bertux) [reply packages/payment-processor/src/payment/encoder-payment.ts:33]: OK I'm preparing it to handle it separately in the API backend.

Comment (bertux) [reply packages/payment-processor/src/payment/encoder-payment.ts:33]: Replaced by #896

**Pull request closed without merging**
